pkg/restmapper/lazyrestmapper.go

```diff
@@ -89,14 +168,14 @@ func fetchGroupVersionResources(d discovery.DiscoveryInterface, apiGroups []*met
 // Mainly replicate ServerGroupsAndResources function from the client-go. The difference is that this function takes
 // a function of the GroupFilterPredicate type as an argument for filtering out unwanted groups.
 // ref: https://github.com/kubernetes/kubernetes/blob/a84d877310ba5cf9237c8e8e3218229c202d3a1e/staging/src/k8s.io/client-go/discovery/discovery_client.go#L383
-func filteredServerGroupsAndResources(d discovery.DiscoveryInterface, groupFilterPredicate GroupFilterPredicate) ([]*metav1.APIGroup, []*metav1.APIResourceList, error) {
+func (m *LazyRESTMapper) filteredServerGroupsAndResources(d discovery.DiscoveryInterface) ([]*metav1.APIGroup, []*metav1.APIResourceList, error) {
 	sgs, err := d.ServerGroups()
 	if sgs == nil {
 		return nil, nil, err
 	}
 	resultGroups := []*metav1.APIGroup{}
 	for i := range sgs.Groups {
-		if groupFilterPredicate(&sgs.Groups[i]) {
+		if m.isGroupKnown((&sgs.Groups[i]).Name) {
```

Comment:
I have a feeling, that for every new group will happen `len(KnownGroups)` number of requests, since it makes a request for every group down below.